addons/crm/static/tests/forecast_kanban_tests.js

```diff
@@ -0,0 +1,260 @@
+                            <div><field name="name"/></div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            archs: {
+                "lead,false,search": `
+                    <search>
+                        <filter name="forecast" string="Forecast" context="{'forecast_filter':1}"/>
+                        <filter name='groupby_date' context="{'group_by':'date'}"/>
+                        <filter name='groupby_datetime' context="{'group_by':'datetime'}"/>
+                    </search>`
+            },
+            data: {
+                lead: {
+                    fields: {
+                        name: {string: 'Name', type: 'char'},
+                        date: {string: "Date Field", type: 'date'},
+                        datetime: {string: "Datetime Field", type: 'datetime'},
+                    },
+                    records: [
+                        {id: 1, name: 'Lead 1', date: '2021-01-01', datetime: '2021-01-01 00:00:00'},
+                        {id: 2, name: 'Lead 2', date: '2021-01-20', datetime: '2021-01-20 00:00:00'},
+                        {id: 3, name: 'Lead 3', date: '2021-02-01', datetime: '2021-02-01 00:00:00'},
+                        {id: 4, name: 'Lead 4', date: '2021-02-20', datetime: '2021-02-20 00:00:00'},
+                        {id: 5, name: 'Lead 5', date: '2021-03-01', datetime: '2021-03-01 00:00:00'},
+                        {id: 6, name: 'Lead 6', date: '2021-03-20', datetime: '2021-03-20 00:00:00'},
+                    ],
+                },
+            },
+            services: { 'forecast': ForecastService },
+            model: 'lead',
+            View: ForecastKanbanView,
+            viewOptions: {
+                context: {
+                    search_default_forecast: true,
+                    search_default_groupby_date: true,
+                    forecast_field: 'date',
+                },
+            },
+            groupBy: ['date'],
+        };
+        this.unPatchDate = await testUtils.mock.patchDate(2021, 1, 10, 0, 0, 0);
+    },
+    afterEach: async function () {
+        await this.unPatchDate();
+    },
+
+}, function () {
+    QUnit.test("filter out every records before the start of the current month with forecast_filter", async function (assert) {
+        assert.expect(14);
+
+        var kanban = await testUtils.createView(this.testKanbanView);
+
+        // the filter is active
+        assert.strictEqual(kanban.$('.o_kanban_group').length, 2, "There should be 2 columns");
+        assert.strictEqual(kanban.$('.o_kanban_group:nth-child(1) .o_kanban_record').length, 2,
+                        "1st column February should contain 2 record");
+        assert.strictEqual(kanban.$('.o_kanban_group:nth-child(2) .o_kanban_record').length , 2,
+                        "2nd column March should contain 2 records");
+        kanban.destroy();
+
+        // remove the filter
```

Comment:
If I understood well, could be rephrased to ``// remove filter on which ``forecast_filter`` context key is used, removing domain with date field being un future`` ?

Reply:
What's the difference between this test and the one at line 110 ?

Reply:
This is the same test, but on a `datetime` field instead of a `date` field, I'll add a comment ^^